src/Iterator/TypedIterator.php

```diff
@@ -37,10 +38,8 @@ public function __construct(Iterator $iterator, ?callable $getType = null)
              * @param mixed $variable
              */
             static function ($variable): string {
-                $type = gettype($variable);
-
-                if ('object' !== $type) {
-                    return $type;
+                if (!is_object($variable)) {
```

Comment:
#AutoFacepalm

Reply:
turns out the previous comparison is not interpreted by PHPStan in the same way that the dedicated function comparison is

Reply:
Great finding!